libs/MailMessage/src/MailMessageBuilder.cpp

```diff
@@ -0,0 +1,68 @@
+#include "MailMessageBuilder.h"
+
+#include "MailAddress.h"
+#include "MailMessage.h"
+#include "MailAttachment.h"
+
+namespace ISXMM
+{
+MailMessageBuilder& MailMessageBuilder::set_from(const std::string& email, const std::string& name)
+{
+    m_from = { email, name };
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::add_to(const std::string& email, const std::string& name)
+{
+    m_to.push_back({ email, name });
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::add_cc(const std::string& email, const std::string& name)
+{
+    m_cc.push_back({ email, name });
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::add_bcc(const std::string& email, const std::string& name)
+{
+    m_bcc.push_back({ email, name });
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::set_subject(const std::string& subject)
+{
+    m_subject = subject;
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::set_body(const std::string& body)
+{
+    m_body = body;
+    return *this;
+}
+
+MailMessageBuilder& MailMessageBuilder::add_attachment(const std::string &path)
+{
+    try
+    {
+        m_attachments.push_back({ path }); 
+    }
+    catch(const std::exception& e)
+    {
+        throw e;
+    }
+    
+    return *this;
+}
+
+MailMessage MailMessageBuilder::Build()
```

Comment:
Well, I just realised that probably in Rust code I reviewed yesterday and here you are using builder design pattern. I'm not against it, I was just imaging its use cases for more complex classes.